lib/model/query/submissions.js

```diff
@@ -90,13 +95,39 @@ with logical as (update submissions set "reviewState"='edited', "updatedAt"=cloc
 , def as (
   ${_defInsert(deprecated.submissionId, partial, form.def.id, actorId, null, deviceId, userAgent)}
 )
-select ${_unjoiner.fields} from (
+select "submissions"."id" as "id"
+     , "submissions"."formId" as "formId"
+     , "submissions"."instanceId" as "instanceId"
+     , "submissions"."submitterId" as "submitterId"
+     , "submissions"."deviceId" as "deviceId"
+     , "submissions"."createdAt" as "createdAt"
+     , "submissions"."updatedAt" as "updatedAt"
+     , "submissions"."reviewState" as "reviewState"
+     , "submissions"."userAgent" as "userAgent"
+     , "submissions"."draft" as "draft"
+     , "submissions"."deletedAt" as "deletedAt"
+     , "submission_defs"."id" as "submissionDefId"
+     , "submission_defs"."createdAt" as "submissionCreatedAt"
+from (
   select logical.*, upd."userAgent" from logical
   join upd on logical.id = upd."submissionId" and root
 ) submissions
 join def submission_defs on submissions.id = submission_defs."submissionId"
 `)
-    .then(_unjoiner);
+    .then(({ submissionDefId, ...submissionData }) => // TODO/HACK: reassembling this from bits and bobs.
+      new Submission(submissionData, {
+        currentVersion: new Submission.Def({
+          id: submissionDefId,
+          instanceId: partial.instanceId,
+          createdAt: submissionData.submissionCreatedAt,
+          deviceId,
+          userAgent,
+          instanceName: partial.def.instanceName,
+          current: true,
+          submitterId: submissionData.submitterId,
+        }),
+        xml: new Submission.Xml({ xml: partial.xml })
+      }));
```

Comment:
one of the reasons that we do this is that we return Submission object in the response of POST and PUT /submissions REST API, what if we don't do that and just return 201/200 with empty body just like OpenRosa? that would be breaking change but can we put this on deprecation path @lognaturel? That would certainly be performance gain and cleaner code.